react/week4/github-user-searcher-app/src/Components/SearchResulsContainer.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+import SearchResults from './SearchResults';
+import Context from '../DefaultContext'
+import { useContext } from 'react';
+
+const SearchResultsContainer = () => {
+    const [results, setResults] = useState([]);
+    const { setStatus, searchToken, searchType } = useContext(Context);
+    useEffect(() => {
+        if (searchToken !== '') {
+            setStatus('loading');
+            const urlToFetch = searchType === 'namesearch' ?
+                `https://api.github.com/search/users?q=${searchToken}` :
+                `https://api.github.com/search/repositories?q=org%3AHackYourFuture-CPH+${searchToken}`;
+
+            fetch(urlToFetch)
+                .then(response => response.json())
+                .then(result => {
+                    setStatus('');
+                    console.log(result.items)
+                    setResults(result.items)
+
+                })
+                .catch(error => {
+                    setStatus(error + '');
+                    setResults([])
+                })
+        } else {
+            setResults([])
+        }
+    }, [searchToken])
```

Comment:
Your logic is fine but it could be contained in the Provider. So you can have simpler consumers. Think how nice would be to have a hook which returns only what you need from the context. Something like:

`const { results, setSearchToken, loading } = useContext(...)`

Reply:
Amazing to learn best practice